### PR TITLE
Add intial data import method and methods to retrieve graphql schema and database schema

### LINKED ISSUE #36: Add method to retrieve the graphql schema
It can be really useful during test development to know the graphql schema generated from graphback in order to write additional queries/mutations. Because the testx server use apollo server, and apollo server exposes the graphql schema, we could add a method, to the TestxServer API, that read and return it

### PR DESCRIPTION
**Pull request by jhellar**
## Description

Resolves: #36 #18 

## Verification

1. checkout this PR
2. `npm link`
3. `cd <FOLDER_WITH_SCHEMA>`
4. `npx -p graphql-testx print-schema <schema.graphql>`

### CODE REVIEW

**Comment by b1zzu** [comment]
@jhellar nice approach, I see it works really good with our examples, but it may be harder to use for external projects, maybe we can make the print schema a package binary so that we can use `npx testx-print-schema schema.graphql` command, wdyt?

**Comment by b1zzu** [comment]
also, the `typeDefs` generated by the backend should be already the schema https://github.com/aerogear/graphql-testx/blob/d7d1272165f4023635a2037c5be8273d19ae4e3e/src/index.ts#L70
so probably there is no need to retrieve it from the running server

**Comment by jhellar** [comment]
@b1zzu that makes sense, it should be now possible to use it with `npx`.

File: bin/printSchema.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import { TestxServer } from "../src";
+
+(async () => {
+  const schemaFile = process.argv[2];
+  const schema = readFileSync(schemaFile, 'utf8');
+
+  const server = new TestxServer(schema);
+  await server.start();
+
+  const response = await axios.post(server.url(), { query: introspectionQuery });
```
Comment (b1zzu) [inline bin/printSchema.ts]: What about moving this line in a method, we could call it `await server.getGraphqlSchema()`, wdyt?

File: package.json [hunk 2]
```diff
@@ -6,13 +6,15 @@
...
-    "lint": "tslint '*(test|src)/**/*.ts'"
+    "lint": "tslint '*(test|src)/**/*.ts'",
+    "prepare": "npm run build"
   },
   "devDependencies": {
     "@types/express": "4.17.2",
     "@types/node": "12.12.11",
     "apollo-boost": "0.4.4",
     "ava": "2.4.0",
+    "axios": "^0.19.0",
```
Comment (b1zzu) [inline package.json]:
```suggestion
    "axios": "0.19.0",
```
we try to avoid fix the package versions and avoid to use `^`

File: src/index.ts [hunk 3]
```diff
@@ -54,6 +54,25 @@ export class TestxServer {
     return this.serverUrl;
   }
 
+  public async getDbSchema() {
```
Comment (b1zzu) [inline src/index.ts]:
```suggestion
  public async getDatabaseSchema() {
```
I would personally prefer extensive names

File: src/index.ts [hunk 4]
```diff
@@ -54,6 +54,25 @@ export class TestxServer {
...
+    for (const table of tables) {
+      schema[table] = Object.keys(await this.dbConnection('task').columnInfo());
+    }
+    return schema;
+  }
+
+  public async setData(data) {
+    const tables = await this.getDbTables();
+    for (const table of tables) {
+      await this.dbConnection(table).del();
```
Comment (b1zzu) [inline src/index.ts]: Myabe we can use the `cleanDatabase()` method

**Review by b1zzu — COMMENTED** [review]
Cool, I have just some small changes

File: bin/printSchema.ts (hunk 1 above)
Comment (jhellar) [reply bin/printSchema.ts]: moved

File: package.json (hunk 2 above)
Comment (jhellar) [reply package.json]: not needed anymore

File: src/index.ts (hunk 3 above)
Comment (jhellar) [reply src/index.ts]: fixed

**Comment by jhellar** [comment]
@b1zzu should be ready for another review

**Comment by b1zzu** [comment]
I think that by adding the `bin` folder to the typescript compiler we have broken the `main` option
https://github.com/aerogear/graphql-testx/blob/9d62f1142a8d660424770f673b8de3b92e6eca15/package.json#L4
and maybe also the `type` option

**Comment by jhellar** [comment]
@b1zzu yeah, that is right, will fix that

**Review by b1zzu — APPROVED** [review]

**Pull request merged**